demosplan/DemosPlanCoreBundle/Logic/EntityContentChangeService.php

```diff
@@ -218,7 +220,9 @@ public function diffArrayAndObject(array $preUpdateArray, CoreEntity $incomingUp
             if ($preUpdateValue instanceof Collection) {
                 // getOriginalEntityData() seems to be ignore n:m association.
                 // use getSnapshot() to get "pre update" data
+                /* @var AbstractLazyCollection $preUpdateValue */
                 $preUpdateValue->initialize();
+                /** @var PersistentCollection $preUpdateValue */
                 $preUpdateValue = $preUpdateValue->getSnapshot();
             }
```

Comment:
This seems to be wrong. Why should the type of `$preUpdateValue` have changed form AbstractLazyCollection to PersistentCollection?

It is possible that `$preUpdateValue` is a PersistentCollection?
If so, it could be done by change the if by checking for instance of PersistentCollection.
PersistentCollection have a `initialize()` as well as a `getSnapshot()`.

Reply:
done in [685364f](https://github.com/demos-europe/demosplan-core/pull/2960/commits/685364f97784d85dd520137c6bcee9f67c93e07c)

Reply:
done in [c3a0a86](https://github.com/demos-europe/demosplan-core/pull/2960/commits/c3a0a86cc653067d1e739215682c0478eed7f83b)